server/src/main/java/org/elasticsearch/repositories/blobstore/FileRestoreContext.java

```diff
@@ -173,20 +182,8 @@ public void restore(SnapshotFiles snapshotFiles, Store store, ActionListener<Voi
         }
     }
 
-    private void afterRestore(SnapshotFiles snapshotFiles, Store store, StoreFileMetadata restoredSegmentsFile) {
-        try {
-            if (isSearchableSnapshotStore(store.indexSettings().getSettings()) == false) {
-                Lucene.pruneUnreferencedFiles(restoredSegmentsFile.name(), store.directory());
```

Comment:
This code does not look to do anything, and is actually in the way as it opens the Lucene index before it's converted. No tests are breaking with this code, and it's unclear whether it still serves a purpose. If we want to be extra defensive, we could move it to StoreRecovery so that it still runs after the index conversion.

Reply:
Yes, it looks like we clean up according to the snapshot contents anyway - this second cleanup was added later on in https://github.com/elastic/elasticsearch/pull/8969 but there's no indication why it was needed. I think you're right anyway, no need to open the index here.